src/generic/graphicc.cpp

```diff
@@ -527,7 +527,22 @@ class WXDLLIMPEXP_CORE wxCairoContext : public wxGraphicsContext
     int m_mswStateSavedDC;
 #endif
 #ifdef __WXGTK__
+    // This factor must be applied to the font before actually using it, for
+    // consistency with the text drawn by GTK itself.
     float m_fontScalingFactor;
+
+    // Function returning Pango font description for the given font scaled by
+    // the font scaling factor if necessary.
+    const PangoFontDescription* GetScaledFontDescription(const wxFont& font) const
+    {
+        if ( m_fontScalingFactor == 1.0f )
+        {
+            return font.GetNativeFontInfo()->description;
+        }
+
+        // We really need to scale the font, so do it.
+        return font.Scaled(m_fontScalingFactor).GetNativeFontInfo()->description;
```

Comment:
Out of pure curiosity, is this really more efficient or the same? wxFont::Scaled creates a new font on the stack anyway, and then copies it in the return (may be optimised).

I've also got confused by the scope of the description string to which the returned pointer points. It seems to work fine, so I am probably getting my thoughts twisted ;)

Reply:
The idea is that it avoids the call to `Scaled()` completely in the common case when the font scaling factor is 1 and while I admit I haven't measured it, I suspect it's not completely trivial: in your original version, instead of just using the font we already have in this case, we created a copy of it (cheap, due to ref counting, but still not free) and scaled it, which created an entirely new font. You've corrected the latter later, of course, but I don't think we want to modify `Scale()`, IMO it would be really better to check for this in `SetFractionalPointSize()` itself, as I wrote on wx-dev. But I didn't have time to do this yet and I didn't want to merge something that resulted in negative effects in the common case. So I decided that this was a nice alternative which avoid the problem completely.

However it seems I've introduced a bad bug when doing it :-( `PangoFontDescription` here is probably indeed deallocated when the temporary font is, thanks for noticing this! I'll fix this a.s.a.p.